roles/confluent.kafka_connect/tasks/main.yml

```diff
@@ -103,3 +103,10 @@
     name: "{{kafka_connect.service_name}}"
     enabled: "{{kafka_connect.systemd.enabled}}"
     state: "{{kafka_connect.systemd.state}}"
+
+- name: Register connector configs and remove deleted connectors
+  confluent_connectors:
+    connect_url: "{{kafka_connect_http_protocol}}://{{kafka_connect.hosts[0]}}:{{kafka_connect_rest_port}}/connectors"
+    active_connectors: "{{ kafka_connect_connectors }}"
+  when: kafka_connect_connectors is defined and kafka_connect.hosts|count > 0 
```

Comment:
Nice change, i dont think we'd ever get to this line if kafka_connect.hosts|count = 0

Reply:
I'm still testing those changes (forgot that the PR would be updated with each commit). I'm getting 'dict object' has no attribute 'hosts'. I'll confirm when it's working fine.

Reply:
ya i think it should be something like this `"{{ groups['kafka_connect'][0] }}"`

Reply:
Yes, this seems to be fine now.

Reply:
I'll be testing on the 5.4 soon, I need to rebuild my test cluster...